### DISCUSSION TITLE
boxcox transform use?

### DISCUSSION THREAD

**Discussion by cdeterman** [online-ml/river#1190] [General]
In the nowcasting example [here](https://github.com/online-ml/river/blob/main/docs/examples/building-a-simple-nowcasting-model.ipynb) it is mentioned that boxcox transforms are often used in these types of problems.  How would you envision it working in an example with river?  The `TargetStandardScaler` is straightforward but I don't see how to do the 'rolling' box cox transform to keep the lambda value for doing the inverse box cox.  Any recommendations?

**Comment by MaxHalford** [comment]
Hey, sorry I missed this discussion.

You could pass the appropriate functions to `TargetTransformRegressor`, using formulas [from Wikipedia](https://www.wikiwand.com/fr/Transform%C3%A9e_de_Box-Cox):

```py
lam = 3
box_cox_target_transformer = TargetTransformRegressor(
    regressor,
    func=lambda x: (x ** lam - 1) / lam,
    inverse_func=lambda y: y * lam ** 1 / lam + 1
)
```

Note that I haven't tested this code.
